src/include/seq_util.h

```diff
@@ -0,0 +1,137 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2018 SUSE LINUX GmbH
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ * @author Jesse Williamson <jwilliamson@suse.de>
+ *
+*/
+
+#ifndef CEPH_SEQ_UTIL
+ #define CEPH_SEQ_UTIL 1
+
+#include <iterator>
+
+#include <boost/tti/tti.hpp>
+
+namespace ceph::util::detail {
+
+BOOST_TTI_HAS_MEMBER_FUNCTION(push_front)
+
+} // namespace ceph::util::detail
+
+// splice_copy():
+//  - This operation differs from std::list<>::splice() in that it makes
+//  copies rather than moving data, hence the name splice_copy().
+namespace ceph::util {
+
+template <typename SeqT0, typename SeqT1>
+void splice_copy(SeqT0& dest, typename SeqT0::iterator pos, 
+                 const SeqT1& src)
+{
+ dest.insert(pos, std::begin(src), std::end(src));
+}
+
+template <typename SeqT0, typename SeqT1>
+void splice_copy(SeqT0& dest, typename SeqT0::iterator pos, 
+                 const SeqT1&& src)
```

Comment:
What is this for?

Reply:
It's in order to allow temporaries (or any rvalue reference) to be acceptable parameters to the function, so you can write:
vector xsv { 4, 3, 2, 1 };
ceph::util::splice_copy(xsv, end(xsv), list { 1, 2, 3, 4 });